tests/storage/datavolume.go

```diff
@@ -381,6 +383,56 @@ var _ = SIGDescribe("DataVolume Integration", func() {
 					return vm.Status.PrintableStatus
 				}, 180*time.Second, 2*time.Second).Should(Equal(v1.VirtualMachineStatusStopped))
 			})
+
+			It("should accurately aggregate DataVolume conditions from many DVs", func() {
+				sc, exist := libstorage.GetRWOFileSystemStorageClass()
+				if !exist {
+					sc, exist = libstorage.GetRWXFileSystemStorageClass()
+					if !exist {
+						Skip("Skip test when FileSystem storage is not present")
+					}
+				}
+
+				dataVolume1 := libdv.NewDataVolume(
+					libdv.WithRegistryURLSource(cd.DataVolumeImportUrlForContainerDisk(cd.ContainerDiskAlpine)),
+					libdv.WithPVC(libdv.PVCWithStorageClass(sc)),
+				)
+				dataVolume2 := libdv.NewDataVolume(
+					libdv.WithRegistryURLSource(cd.DataVolumeImportUrlForContainerDisk(cd.ContainerDiskAlpine)),
+					libdv.WithPVC(libdv.PVCWithStorageClass(sc)),
+				)
+
+				vmi := tests.NewRandomVMIWithDataVolume(dataVolume1.Name)
+
+				By("requiring an additional DataVolume")
+				diskName := "disk1"
+				vmi = libstorage.AddDataVolumeDisk(vmi, diskName, dataVolume2.Name)
+
+				vmSpec := tests.NewRandomVirtualMachine(vmi, true)
+
+				vm, err := virtClient.VirtualMachine(vmSpec.Namespace).Create(context.Background(), vmSpec)
+				Expect(err).ToNot(HaveOccurred())
+
+				Eventually(ThisVM(vm), 180*time.Second, 2*time.Second).Should(havePrintableStatus(v1.VirtualMachineStatusPvcNotFound))
+
+				By("creating the first DataVolume")
+				dataVolume1, err = virtClient.CdiClient().CdiV1beta1().DataVolumes(vmSpec.Namespace).Create(context.Background(), dataVolume1, metav1.CreateOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				defer libstorage.DeleteDataVolume(&dataVolume1)
+
+				By("ensuring that VMI and VM are reporting VirtualMachineInstanceDataVolumesReady=False")
+				Eventually(ThisVMI(vmi), 180*time.Second, 2*time.Second).Should(haveDataVolumesReadyConditionStatus(k8sv1.ConditionFalse))
+				Eventually(ThisVM(vm), 180*time.Second, 2*time.Second).Should(haveDataVolumesReadyConditionStatus(k8sv1.ConditionFalse))
+
+				By("creating the second DataVolume")
+				dataVolume2, err = virtClient.CdiClient().CdiV1beta1().DataVolumes(vmSpec.Namespace).Create(context.Background(), dataVolume2, metav1.CreateOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				defer libstorage.DeleteDataVolume(&dataVolume2)
```

Comment:
This is redundant.

Reply:
See https://github.com/kubevirt/kubevirt/blob/main/tests/testsuite/namespace.go#L186-L188.
Ideally, tests should not care about cleaning up to write neat tests.

Again, I acknowledge that we did not clean up the whole test suite and I do insist on discontinuing accepting this pattern.